font/src/darwin/mod.rs

```diff
@@ -454,6 +454,19 @@ impl Font {
     }
 
     pub fn get_glyph(&self, character: char, _size: f64, use_thin_strokes: bool) -> Result<RasterizedGlyph, Error> {
+        // Render custom symbols for underline and beam cursor
+        if character == super::UNDERLINE_CURSOR_CHAR {
+            let descent = -(self.ct_font.descent() as i32);
+            let width = self.glyph_advance('0') as i32;
+            return super::get_underline_cursor_glyph(descent, width);
+        } else if character == super::BEAM_CURSOR_CHAR {
+            let metrics = self.metrics();
```

Comment:
Might be nice to have a comment suggesting the _intent_ of these calculations in case someone needs to come work on them. Ditto for above.